tests/jobs/python/pi_with_secret.py

```diff
@@ -0,0 +1,48 @@
+import os
+import sys
+from random import random
+from operator import add
+
+from pyspark.sql import SparkSession
+
+def check_secret(secret_name, secret_content):
+    '''
+    Make sure the extra secret envvar and secret file show up in driver.
+    '''
+    envvar_content = os.environ.get(secret_name)
+    if envvar_content != secret_content:
+        print("Unexpected contents in secret envvar, found: {} expected: {}".format(envvar_content, secret_content))
+        exit(1)
+
+    file_content = open(secret_name, 'r').read()
+    if file_content != secret_content:
+        print("Unexpected contents in secret file, found: {} expected: {}".format(file_content, secret_content))
+        exit(1)
+
+
+if __name__ == "__main__":
+    """
+        Usage: pi [partitions] [secret] [secret content]
+        Checks for the given env-based and file-based driver secret.
+        Then calculates the value of pi.
+    """
+
+    check_secret(sys.argv[2], sys.argv[3])
```

Comment:
does this also check the secret in the executors or just the driver?

Reply:
I didn't bother adding a test for executor secrets because the CLI code (wrt TLS) is the same for driver and executor secrets.